prettier-plugin-svelte doesn't work with eslint-plugin-prettier

<!-- If you're reporting a bug, please provide the following information to make it easier to triage your issue. If you don't provide this information, your issue may be closed. If a question isn't applicable to your report, feel free to skip it/delete it. -->

**What version of `eslint` are you using?**
v8.17.0
**What version of `prettier` are you using?**
2.7.1
**What version of `eslint-plugin-prettier` are you using?**
4.0.0
**Please paste any applicable config files that you're using (e.g. `.prettierrc` or `.eslintrc` files)**
```json
{
  "root": true,
  "env": {
    "node": true,
    "es2021": true,
    "mocha": true,
    "browser": true
  },
  "ignorePatterns": ["dist", "**/*.html"],
  "extends": ["eslint:recommended"],
  "plugins": ["prettier", "svelte3"],
  "rules": {
    "strict": 0,
    "no-console": 0,
    "max-len": [
      "error",
      {
        "code": 120,
        "ignoreComments": true
      }
    ],
    "prettier/prettier": [
      "warn",
      {
        "printWidth": 120,
        "tabWidth": 2,
        "bracketSpacing": false,
        "trailingComma": "none",
        "arrowParens": "avoid",
        "pluginSearchDirs": ["."],
        "plugins": ["prettier-plugin-svelte"]
      }
    ]
  },
  "overrides": [
    {
      "files": ["*.svelte"],
      "processor": "svelte3/svelte3"
    }
  ],
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 2021
  }
}
```
**What source code are you linting?**
```svelte
<script>
  /* TODO:
   * implementar links para editar con diálogos y definir si lo que se va a editar
   * es un string, un text o html
   */
  import {onMount} from "svelte";
  import {writable} from "svelte/store";
  function createRef(val) {
    const ref = writable(val);
    return [ref, ref.set];
  }
  // @ts-ignore
  import {requestBase} from "request-strategy";
  
  const request = requestBase(createRef, cb => onMount(() => cb), true);
  export let debug;
  export let locale;
  export let allowTranslations;
  export let showTranslations;
  export let context;
  export let subcontext;
  export let text;
  export let baseurl;
  let input;
  let mounted = false;
  let firstRequest = true;
  let result, error, mutate;
  // eslint-disable-next-line max-len
  $: url =
    (baseurl ? baseurl : "") +
    `/APP/AAA/Translate.translate?xhr=true&context=${context}&sub-context=${subcontext}&text=${text}&locale=${$locale}`;
  onMount(() => {
    ({result, error, mutate} = request(url, "SWR", undefined, true));
    mounted = true;
  });
  $: {
    if ($locale && mounted && firstRequest) {
      firstRequest = false;
    } else if ($locale && mounted && !firstRequest) {
      mutate(url);
    }
  }
  $: width = "min(calc(" + ($result?.text?.length || 10) + "ch + 5ch), 100%)";
  async function onChange() {
    // @ts-ignore
    const cookies = new Map(document.cookie.split("; ").map(c => c.split("=")));
    const xsrf = cookies.get("XSRF-TOKEN");
    const response = await fetch(url, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
        "XSRF-TOKEN": xsrf
      },
      body: JSON.stringify({translation: input.value}),
      mode: "cors",
      credentials: "include"
    });
    const res = await response.json();
    if (res.status === "ok") {
      // await Promise.delay(500);
      //TODO forzar el mutate
      await mutate(undefined, true);
    }
  }
</script>
<svelte:options tag={null}/>

{#if $error}
  Error loading translation
{:else if !$result}
  Cargando...
{:else}
  <span class={$showTranslations ? $result.status.toLowerCase() : ""}>
    {#if $allowTranslations}
      <input 
value={$result.text} 
bind:this={input}
on:change={onChange}
on:click|preventDefault|stopPropagation
style:width
            />
    {:else}
      {$result.text}
    {/if}
  </span>
{/if}
{#if typeof debug !== "undefined"}
  <div> locale {$locale}</div>
  <div> allowTranslations {$allowTranslations}</div>
  <div> showTranslations {$showTranslations}</div>
  <div> context {context}</div>
  <div> subContext {subcontext}</div>
  <div> text {text}</div>
  {#if $result}
    <div> result {$result.text}</div>
  {/if}
  <div> error {$error}</div>
{/if}
<style>
.ok {
  background-color: green;
  color: white;
  border: dotted;
}
.failed {
  background-color: red;
  color: white;
  border: dotted;
}
.waiting {
  background-color: yellow;
  border: dotted;
}
.generic {
  background-color: lightblue;
  border: dotted;
}

</style>
```
**What did you expect to happen?**
I want to be able to format svelte files just like prettier do.
 
**What actually happened?**
I'm trying to add a plugin to prettier to allow svelte files to be linted and formated but, I get this output:
```sh
pnpm eslint src/lib/Translations.svelte

/home/jcmunoz/Dropbox/Develop/node.js/itdfw-monorepo/web-components/itd-wc-translate-svelte/src/lib/Translations.svelte
  66:26  warning  Replace `⏎);(` with `;`  prettier/prettier
  68:12  warning  Replace `⏎);(` with `;`  prettier/prettier
  70:19  warning  Replace `⏎);(` with `;`  prettier/prettier
  73:69  warning  Replace `⏎);(` with `;`  prettier/prettier
  74:28  warning  Replace `⏎);(` with `;`  prettier/prettier
  76:20  warning  Replace `⏎);(` with `;`  prettier/prettier
  77:17  warning  Replace `⏎);(` with `;`  prettier/prettier
  78:20  warning  Replace `⏎);(` with `;`  prettier/prettier
  83:20  warning  Replace `⏎);(` with `;`  prettier/prettier
  87:34  warning  Replace `⏎);(` with `;`  prettier/prettier
  88:24  warning  Replace `⏎);(` with `;`  prettier/prettier
  89:46  warning  Replace `⏎);(` with `;`  prettier/prettier
  90:44  warning  Replace `⏎);(` with `;`  prettier/prettier
  91:25  warning  Replace `⏎);(` with `;`  prettier/prettier
  92:31  warning  Replace `⏎);(` with `;`  prettier/prettier
  93:19  warning  Replace `⏎);(` with `;`  prettier/prettier
  94:15  warning  Replace `⏎);(` with `;`  prettier/prettier
  95:31  warning  Replace `⏎);(` with `;`  prettier/prettier
  97:22  warning  Insert `;`               prettier/prettier

✖ 19 problems (0 errors, 19 warnings)
  0 errors and 19 warnings potentially fixable with the `--fix` option.
```
If I use prettier alone, it formats the file correctly, using the same configuration as in the .eslintrc
```json
{
  "printWidth": 120,
  "tabWidth": 2,
  "bracketSpacing": false,
  "trailingComma": "none",
  "arrowParens": "avoid",
  "plugins": ["prettier-plugin-svelte"],
}
```
```sh
pnpm prettier  src/lib/Translations.svelte
```
```svelte
<svelte:options tag={null} />

<script>
  /* TODO:
   * implementar links para editar con diálogos y definir si lo que se va a editar
   * es un string, un text o html
   */
  import {onMount} from "svelte";
  import {writable} from "svelte/store";
  function createRef(val) {
    const ref = writable(val);
    return [ref, ref.set];
  }
  // @ts-ignore
  import {requestBase} from "request-strategy";

  const request = requestBase(createRef, cb => onMount(() => cb), true);
  export let debug;
  export let locale;
  export let allowTranslations;
  export let showTranslations;
  export let context;
  export let subcontext;
  export let text;
  export let baseurl;
  let input;
  let mounted = false;
  let firstRequest = true;
  let result, error, mutate;
  // eslint-disable-next-line max-len
  $: url =
    (baseurl ? baseurl : "") +
    `/APP/AAA/Translate.translate?xhr=true&context=${context}&sub-context=${subcontext}&text=${text}&locale=${$locale}`;
  onMount(() => {
    ({result, error, mutate} = request(url, "SWR", undefined, true));
    mounted = true;
  });
  $: {
    if ($locale && mounted && firstRequest) {
      firstRequest = false;
    } else if ($locale && mounted && !firstRequest) {
      mutate(url);
    }
  }
  $: width = "min(calc(" + ($result?.text?.length || 10) + "ch + 5ch), 100%)";
  async function onChange() {
    // @ts-ignore
    const cookies = new Map(document.cookie.split("; ").map(c => c.split("=")));
    const xsrf = cookies.get("XSRF-TOKEN");
    const response = await fetch(url, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
        "XSRF-TOKEN": xsrf
      },
      body: JSON.stringify({translation: input.value}),
      mode: "cors",
      credentials: "include"
    });
    const res = await response.json();
    if (res.status === "ok") {
      // await Promise.delay(500);
      //TODO forzar el mutate
      await mutate(undefined, true);
    }
  }
</script>

{#if $error}
  Error loading translation
{:else if !$result}
  Cargando...
{:else}
  <span class={$showTranslations ? $result.status.toLowerCase() : ""}>
    {#if $allowTranslations}
      <input
        value={$result.text}
        bind:this={input}
        on:change={onChange}
        on:click|preventDefault|stopPropagation
        style:width
      />
    {:else}
      {$result.text}
    {/if}
  </span>
{/if}
{#if typeof debug !== "undefined"}
  <div>locale {$locale}</div>
  <div>allowTranslations {$allowTranslations}</div>
  <div>showTranslations {$showTranslations}</div>
  <div>context {context}</div>
  <div>subContext {subcontext}</div>
  <div>text {text}</div>
  {#if $result}
    <div>result {$result.text}</div>
  {/if}
  <div>error {$error}</div>
{/if}

<style>
  .ok {
    background-color: green;
    color: white;
    border: dotted;
  }
  .failed {
    background-color: red;
    color: white;
    border: dotted;
  }
  .waiting {
    background-color: yellow;
    border: dotted;
  }
  .generic {
    background-color: lightblue;
    border: dotted;
  }
</style>
```
I don't know if my configuration is wrong or if the eslint-plugin-prettier can't load other plugins.

Regards.

--- a/.changeset/odd-eagles-breathe.md
+++ b/.changeset/odd-eagles-breathe.md
@@ -1,0 +1,9 @@
+---
+'eslint-plugin-prettier': minor
+---
+
+feat: support svelte out of box
+
+close #472, close #482
+
+We recommend to use [`eslint-plugin-svelte`](https://github.com/ota-meshi/eslint-plugin-svelte) instead of [`eslint-plugin-svelte3`](https://github.com/sveltejs/eslint-plugin-svelte3).

--- a/eslint-plugin-prettier.js
+++ b/eslint-plugin-prettier.js
@@ -187,19 +187,33 @@ module.exports = {
               // by default.
               // Related ESLint plugins are:
               // 1. `eslint-plugin-graphql` (replacement: `@graphql-eslint/eslint-plugin`)
-              // 2. `eslint-plugin-markdown@1` (replacement: `eslint-plugin-markdown@2+`)
-              // 3. `eslint-plugin-html`
+              // 2. `eslint-plugin-html`
+              // 3. `eslint-plugin-markdown@1` (replacement: `eslint-plugin-markdown@2+`)
+              // 4. `eslint-plugin-svelte3` (replacement: `eslint-plugin-svelte@2+`)
               const parserBlocklist = [null, 'markdown', 'html'];
 
               let inferParserToBabel = parserBlocklist.includes(inferredParser);
 
-              if (
+              switch (inferredParser) {
                 // it could be processed by `@graphql-eslint/eslint-plugin` or `eslint-plugin-graphql`
-                inferredParser === 'graphql' &&
-                // for `eslint-plugin-graphql`, see https://github.com/apollographql/eslint-plugin-graphql/blob/master/src/index.js#L416
-                source.startsWith('ESLintPluginGraphQLFile`')
-              ) {
-                inferParserToBabel = true;
+                case 'graphql': {
+                  if (
+                    // for `eslint-plugin-graphql`, see https://github.com/apollographql/eslint-plugin-graphql/blob/master/src/index.js#L416
+                    source.startsWith('ESLintPluginGraphQLFile`')
+                  ) {
+                    inferParserToBabel = true;
+                  }
+                  break;
+                }
+                // it could be processed by `@ota-meshi/eslint-plugin-svelte`, `eslint-plugin-svelte` or `eslint-plugin-svelte3`
+                case 'svelte': {
+                  // The `source` would be modified by `eslint-plugin-svelte3`
+                  if (!context.parserPath.includes('svelte-eslint-parser')) {
+                    // We do not support `eslint-plugin-svelte3`,
+                    // the users should run `prettier` on `.svelte` files manually
+                    return;
+                  }
+                }
               }
 
               if (inferParserToBabel) {
@@ -223,6 +237,7 @@ module.exports = {
                 'html',
                 'mdx',
                 'angular',
+                'svelte',
               ];
               if (parserBlocklist.includes(inferredParser)) {
                 return;

--- a/package.json
+++ b/package.json
@@ -48,8 +48,8 @@
     "@changesets/changelog-github": "^0.4.5",
     "@changesets/cli": "^2.23.0",
     "@graphql-eslint/eslint-plugin": "^2.5.0",
+    "@ota-meshi/eslint-plugin-svelte": "^0.34.1",
     "@typescript-eslint/parser": "^5.29.0",
-    "eslint": "^7.32.0",
     "eslint-config-prettier": "^8.5.0",
     "eslint-mdx": "^1.17.0",
     "eslint-plugin-eslint-plugin": "^4.3.0",
@@ -60,6 +60,7 @@
     "mocha": "^9.2.2",
     "patch-package": "^6.4.7",
     "prettier": "^2.7.1",
+    "svelte": "^3.48.0",
     "vue-eslint-parser": "^8.3.0"
   },
   "resolutions": {

--- a/test/fixtures/@ota-meshi/eslint-plugin-svelte/App.svelte
+++ b/test/fixtures/@ota-meshi/eslint-plugin-svelte/App.svelte
@@ -1,0 +1,5 @@
+<script>
+let name = 'world';
+</script>
+
+<h1>Hello {name}!</h1>

--- a/test/fixtures/eslint-plugin-svelte3/App.named-blocks.svelte
+++ b/test/fixtures/eslint-plugin-svelte3/App.named-blocks.svelte
@@ -1,0 +1,5 @@
+<script>
+let name = 'world';
+</script>
+
+<h1>Hello {name}!</h1>

--- a/test/fixtures/eslint-plugin-svelte3/App.svelte
+++ b/test/fixtures/eslint-plugin-svelte3/App.svelte
@@ -1,0 +1,5 @@
+<script>
+let name = 'world';
+</script>
+
+<h1>Hello {name}!</h1>

--- a/test/prettier.js
+++ b/test/prettier.js
@@ -46,6 +46,21 @@ const eslint = new ESLint({
           'mdx/code-block': true,
         },
       },
+      {
+        files: '**/eslint-plugin-svelte3/*.svelte',
+        plugins: ['svelte3'],
+        processor: 'svelte3/svelte3',
+      },
+      {
+        files: '**/eslint-plugin-svelte3/*.named-blocks.svelte',
+        settings: {
+          'svelte3/named-blocks': true,
+        },
+      },
+      {
+        files: '**/@ota-meshi/eslint-plugin-svelte/*.svelte',
+        extends: ['plugin:@ota-meshi/svelte/recommended'],
+      },
     ],
   },
   useEslintrc: false,
@@ -158,8 +173,8 @@ atGraphqlEslintRuleTester.run('@graphql-eslint/eslint-plugin', rule, {
   ],
 });
 
-// eslint-plugin-graphql handles literal graphql files by tranforming graphql
-// code with a processor, instead of using a parser. Unfortunatly we cant
+// eslint-plugin-graphql handles literal graphql files by transforming graphql
+// code with a processor, instead of using a parser. Unfortunately we cant
 // specify custom processors in a RuleTester, so instead we have write test code
 // that is the result of eslint-plugin-graphql's processing - this is the
 // ESLintPluginGraphQLFile tagged template literal. See
@@ -210,7 +225,7 @@ mdxRuleTester.run('eslint-plugin-mdx', rule, {
   ],
 });
 
-runFixture('mdx', [
+runFixture('*.mdx', [
   [
     {
       column: 33,
@@ -244,6 +259,43 @@ runFixture('mdx', [
     },
   ],
 ]);
+
+runFixture('@ota-meshi/eslint-plugin-svelte/*.svelte', [
+  [
+    {
+      column: 5,
+      endColumn: 13,
+      endLine: 2,
+      fix: {
+        range: [13, 21],
+        text: 'name =',
+      },
+      line: 2,
+      message: 'Replace `·name·=·` with `name·=`',
+      messageId: 'replace',
+      nodeType: null,
+      ruleId: 'prettier/prettier',
+      severity: 2,
+    },
+    {
+      column: 4,
+      endColumn: 7,
+      endLine: 5,
+      fix: {
+        range: [45, 48],
+        text: '>',
+      },
+      line: 5,
+      message: 'Replace `·>·` with `>`',
+      messageId: 'replace',
+      nodeType: null,
+      ruleId: 'prettier/prettier',
+      severity: 2,
+    },
+  ],
+]);
+
+runFixture('eslint-plugin-svelte3/*.svelte', [[], []]);
 
 // ------------------------------------------------------------------------------
 //  Helpers
@@ -288,12 +340,18 @@ function getPrettierRcJsFilename(dir, file = 'dummy.js') {
   return path.resolve(__dirname, `./prettierrc/${dir}/${file}`);
 }
 
-async function runFixture(name, asserts) {
+/**
+ *
+ * @param {string} pattern
+ * @param {import('eslint').Linter.LintMessage[]} asserts
+ * @returns {Promise<void>}
+ */
+async function runFixture(pattern, asserts) {
   try {
-    const results = await eslint.lintFiles(`test/fixtures/${name}.*`);
+    const results = await eslint.lintFiles([`test/fixtures/${pattern}`]);
     return assert.deepStrictEqual(
-      asserts,
       results.map(({ messages }) => messages),
+      asserts,
     );
   } catch (err) {
     console.error(err);

